File: modules/wem-api/src/test/java/com/enonic/wem/api/NamePrettyfierTest.java
package com.enonic.wem.api;

import org.junit.Test;

import static org.junit.Assert.*;

public class NamePrettyfierTest
{
    @Test
    public void test_ensure_valid_name_spaces_are_trimmed()
    {
        assertEquals( "test-name", NamePrettyfier.prettifyName( " test Name ", true ) );
    }

    @Test
    public void test_ensure_valid_special_chars1_transliterate()
    {
        assertEquals( "ase-kalrot", NamePrettyfier.prettifyName( "Åse Kålrot", true ) );
    }

    @Test
    public void test_ensure_valid_special_chars1()
    {
        assertEquals( "åse-kålrot", NamePrettyfier.prettifyName( "Åse Kålrot", false ) );
    }

    @Test
    public void test_ensure_valid_special_chars3()
    {
        assertEquals( "test-stuff-here", NamePrettyfier.prettifyName( "test/stuff/here ", true ) );
    }

    @Test
    public void test_ensure_valid_special_chars2()
    {
        assertEquals( "t-e-st-cha-r-s-1", NamePrettyfier.prettifyName( "t@e&st^-$cha$r@s $1 ", true ) );
    }

    @Test
    public void removeInvisibleChars()
    {
        assertEquals( "teststuffhere", NamePrettyfier.prettifyName( "Test\u0081Stuff\u0082Here\u0083", true ) );
    }

    @Test(expected = IllegalArgumentException.class)
    public void test_ensure_valid_name_empty()
    {
        assertEquals( "", NamePrettyfier.prettifyName( "", true ) );
    }

    @Test(expected = IllegalArgumentException.class)
    public void test_ensure_valid_name_space()
    {
        assertEquals( "", NamePrettyfier.prettifyName( " ", true ) );
    }

    @Test(expected = IllegalArgumentException.class)
    public void test_ensure_valid_name_null()
    {
        assertEquals( "", NamePrettyfier.prettifyName( null, true ) );
    }

}

File: modules/admin/admin-ui/src/main/resources/web/dev/test/common/js/NamePrettyfierTest.js
describe("api.NamePrettyfierTest", function () {

    describe("tests for api.NamePrettyfier.prettify() function", function () {


        it("should prettify correctly", function () {
            expect(api.NamePrettyfier.prettify(" test Name ")).toBe("test-name");
            expect(api.NamePrettyfier.prettify("?se K?lrot")).toBe("ase-kalrot");
            expect(api.NamePrettyfier.prettify("test/stuff/here ")).toBe("test-stuff-here");
            expect(api.NamePrettyfier.prettify("t@e&st^-$cha$r@s $1 ")).toBe("t-e-st-cha-r-s-1");
            expect(api.NamePrettyfier.prettify("Test\u0081Stuff\u0082Here\u0083")).toBe("teststuffhere");
            expect(api.NamePrettyfier.prettify("?videm.ment_']]")).toBe("evidem.ment");
        });

    });


});